### PR TITLE
Add PRP, and PRF implies PRP

### PR DESCRIPTION
**Pull request by wyatt-howe**
Also fixed minor JS bug when adding new nodes

### CODE REVIEW

File: objects/Edges/PRF_E_implies_E_PRP.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "uid": 3580279351,
+    "source_nodes": [3320321157],
+    "dest_node": 5079247584,
+    "link_type": "E_implies_E",
+    "bibtex_ref": "luby1988construct",
+    "page_number": "Section 1.1",
```
Comment (sarahscheffler) [inline objects/Edges/PRF_E_implies_E_PRP.json]: The actual method is in Section 5 (Page 380)

File: objects/Edges/PRF_E_implies_E_PRP.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "long_description": "If pseudorandom functions exist, then pseudorandom permutations exist.  Starting with a PRF $G : \\{0,1\\}^n \\rightarrow \\{0,1\\}^{n}$.  We can construct a PRP $F : \\{0,1\\}^n \\cross {0,1\\}^s \\rightarrow \\{0,1\\}^{n}$.",
```
Comment (sarahscheffler) [inline objects/Edges/PRF_E_implies_E_PRP.json]: In addition, we should say the actual method:

Specifically, doing three rounds of a Feistel network (originally \cite{feistel1973cryptography}, explained more fully in \cite{schneier1996unbalanced}) with a PRF as the round function will yield a PRP.

File: objects/Citations/Citations.bib [hunk 3]
```diff
@@ -65,3 +65,15 @@ @inproceedings{rompel1990one
...
+  title = {How to Construct Pseudorandom Permutations from Pseudorandom Functions},
+  journal = {SIAM Journal on Computing},
+  volume = {17},
+  number = {2},
+  pages = {373--386},
+  year = {1988},
+  doi = {10.1137/0217022},
+  publisher={SIAM},
+  howpublished="\url{https://pdfs.semanticscholar.org/7d58/9564998287a942149c4fee931852b22d8997.pdf}"
+}
```
Comment (sarahscheffler) [inline objects/Citations/Citations.bib:101]: I added additional references
@article{feistel1973cryptography,
  title={Cryptography and computer privacy},
  author={Feistel, Horst},
  journal={Scientific american},
  volume={228},
  number={5},
  pages={15--23},
  year={1973},
  publisher={JSTOR},
  howpublished="\url{https://www.jstor.org/stable/24923044}"
}

@inproceedings{schneier1996unbalanced,
  title={Unbalanced Feistel networks and block cipher design},
  author={Schneier, Bruce and Kelsey, John},
  booktitle={International Workshop on Fast Software Encryption},
  pages={121--144},
  year={1996},
  organization={Springer},
  howpublished="\url{https://www.schneier.com/academic/paperfiles/paper-unbalanced-feistel.pdf}"
}

File: objects/Edges/PRF_E_implies_E_PRP.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    "submitter": "wyatt-howe",
```
Comment (sarahscheffler) [inline objects/Edges/PRF_E_implies_E_PRP.json:9]: Since I'm adding info, can you add me as a verifier?
`"verifier": "sarahscheffler",`

**Review by sarahscheffler — CHANGES_REQUESTED** [review]
I added a brief description of the method that builds a PRP from a PRF and added the according citations.  Could you add them?

File: objects/Citations/Citations.bib [hunk 5]
```diff
@@ -65,3 +65,15 @@ @inproceedings{rompel1990one
...
+  author = {Luby, Michael and Rackoff, Charles},
+  title = {How to Construct Pseudorandom Permutations from Pseudorandom Functions},
+  journal = {SIAM Journal on Computing},
+  volume = {17},
+  number = {2},
+  pages = {373--386},
+  year = {1988},
+  doi = {10.1137/0217022},
+  publisher={SIAM},
+  howpublished="\url{https://pdfs.semanticscholar.org/7d58/9564998287a942149c4fee931852b22d8997.pdf}"
```
Comment (sarahscheffler) [inline objects/Citations/Citations.bib]: This URL leads to a paper by Pieprzyk.

The paper you're referencing can be found at [https://search.proquest.com/docview/919828131](https://search.proquest.com/docview/919828131)

**Comment by wyatt-howe** [comment]
Changes made – this update should be ready to merge now.  Thanks for correcting what I missed

**Pull request merged**